app/uk/gov/hmrc/gform/sharedmodel/formtemplate/ComponentType.scala

```diff
@@ -316,6 +316,8 @@ object SummaryDisplayWidth extends Enumeration {
 
 sealed trait OptionData extends Product with Serializable {
   def label: SmartString
+  def hint: Option[SmartString]
+  def dynamic: Option[OptionData.Dynamic]
```

Comment:
shouldn't just:
```
def dynamic: Option[FormComponentId]
```
be enough. We always pattern match OptionData and dynamic is always in the same position.

Reply:
`OptionData.Dynamic` is used in quite a few method signatures. I like to see on the first sigh that param is of type  `OptionData.Dynamic` rather than of type `FormComponentId`. 
In the first case reader can immediately guess what that param value is used for without looking into implementation details.